fastapi_opa/auth/auth_saml.py

```diff
@@ -28,9 +33,47 @@ async def authenticate(
         request_args = await self.prepare_request(request)
         auth = await self.init_saml_auth(request_args)
 
-        if "acs" in request.query_params:
-            return await self.assertion_consumer_service(auth, request_args)
+        if 'sso' in request.query_params:
+            logger.debug(datetime.utcnow(), '--sso--')
+            return await self.single_sign_on(auth)
+            # TODO: check below code
+            # If AuthNRequest ID need to be stored in order to later validate it, # do instead
+            # sso_built_url = auth.login()
+            # request.session['AuthNRequestID'] = auth.get_last_request_id()
+            # return redirect(sso_built_url)
+        elif 'sso2' in request.query_params:
+            logger.debug(datetime.utcnow(), '--sso2--')
+            return_to = '%sattrs/' % request.base_url
+            return await self.single_sign_on(auth, return_to)
+        elif "acs" in request.query_params:
+            logger.debug(datetime.utcnow(), '--acs--')
+            return await self.assertion_consumer_service(auth, request_args, request)
         # potentially extend with logout here
+        elif 'slo' in request.query_params:
+            logger.debug(datetime.utcnow(), '--slo--')
+            return await self.single_log_out(auth)
+        elif 'example-read-cookie' in request.query_params:
```

Comment:
This should obviously not be here. It was only an example how to use the session.

Reply:
Deleted